File: Viikko1_teht2/Opiskelija.java
public class Opiskelija {
	//Tietojäsenet alkuun
	private String mNimi;
	private String mOpiskelijanumero;
	private int mOpintopisteet = 0;

	public Opiskelija(){
	System.out.println("vakioparametri opiskelija luotu." );
	mNimi = "Tuntematon";
	mOpiskelijanumero = "xxxxx";
	mOpintopisteet = 0;
}	
	public Opiskelija(String aNimi, String aOpiskelijanumero){
	System.out.println("2 parametrin opiskelija luotu" );
	mNimi = aNimi;
	mOpiskelijanumero = aOpiskelijanumero;
}
	public void setNimi(String aNimi ){
	mNimi = aNimi;
}
	public String getNimi(){
	return mNimi;
}
	public void setOpintopisteet(int aOpintopisteet){
	mOpintopisteet = aOpintopisteet;
} 
	public int getOpintopisteet(){
	return mOpintopisteet;
}
	public void setOpiskelijanumero(String aOpiskelijanumero ){
	mOpiskelijanumero = aOpiskelijanumero;
}
	public String getOpiskelijanumero(){
	return mOpiskelijanumero;
}
	public void tulostatiedot(){
	System.out.println("Nimi: " + mNimi );
 	System.out.println("Opiskelijanumero: " + mOpiskelijanumero); 
 	System.out.println("Opintopisteet: " + mOpintopisteet); 

	}

}

File: viikko2_teht1/viikko2_teht1/Opiskelija.cpp
#include "Opiskelija.h"


Opiskelija::Opiskelija()
{
	cout << "Opiskeljan oletusrakentaja" << endl;
}

Opiskelija::Opiskelija(const string& aNimi, const string& aOpiskelijanumero)
	: mNimi(aNimi), mOpiskelijanumero(aOpiskelijanumero), mOpintopisteet(0)
{
	cout << "Opiskeljan 2 parametrin oletusrakentaja" << endl;
}

Opiskelija::~Opiskelija()
{
	cout << "Opiskelja " << mNimi << " poistettu muistista" << endl;
}

void Opiskelija::setNimi(const string& aNimi)
{
	mNimi = aNimi;
}

void Opiskelija::SetOpiskelijanumero(const string& aOpiskelijanumero)
{
	mOpiskelijanumero = aOpiskelijanumero;
}

void Opiskelija::setOpintopisteet(int aOpintopisteet)
{
	mOpintopisteet = aOpintopisteet;
}

string Opiskelija::getNimi() const
{
	return mNimi;
}

string Opiskelija::getOpiskelijanumero() const
{
	return mOpiskelijanumero;
}

int Opiskelija::getOpintopisteet() const
{
	return mOpintopisteet;
}


void Opiskelija::tulostaTiedot() const
{
	cout << "Opiskelijan nimi: " << mNimi << endl;
	cout << "Opiskelijanumero: " << mOpiskelijanumero << endl;
	cout << "Opintopisteet: " << mOpintopisteet << endl;
}